### PR TITLE
Improve Utf8JsonWriter code coverage

### PR DESCRIPTION
**Pull request by madmir**
* Added test for expected argument exception when property name or value arguments reach MaxTokenSize

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@madmir, can you please resolve the conflicts based on recent `Utf8JsonWriter` API changes? Let me know and I can review the PR then: https://github.com/dotnet/corefx/pull/36961

**Comment by madmir** [comment]
API changes made my previous commit (testing buffer too small and WriteComment methods) obsolete/redundant. Therefore I got rid of it and the latest test is based on how WriteLargeKeyOrValue and WriteLargeKeyValue are already working. It should increase System.Text.Json.ThrowHelper coverage.

**Comment by ahsonkhan** [comment]
> It should increase System.Text.Json.ThrowHelper coverage.

Can you share the code coverage report? Go to the test directory and run: `dotnet msbuild /t:BuildAndTest /p:Coverage=true`

https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md

**Comment by madmir** [comment]
> Can you share the code coverage report? 

[code_coverage_36811.zip](https://github.com/dotnet/corefx/files/3119642/code_coverage_36811.zip)

File: src/System.Text.Json/tests/Utf8JsonWriterTests.cs [hunk 1]
```diff
@@ -3369,6 +3370,82 @@ private static void WriteTooLargeHelper(JsonWriterOptions options, ReadOnlySpan<
             jsonUtf8.Flush();
         }
 
+        [ConditionalTheory(nameof(IsX64))]
+        [OuterLoop]
+        [InlineData(true, true)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(false, false)]
+        public void WriteTooLargeArguments(bool formatted, bool skipValidation)
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/Utf8JsonWriterTests.cs]: This test should run relatively quickly, correct, given the check happens right up front?

I am fine it running in outerloop (given it allocates quite a bit), but just checking that runtime wise it wouldn't cause a significant time cost.

Comment (madmir) [reply src/System.Text.Json/tests/Utf8JsonWriterTests.cs]: Yes. Memory allocation is the reason, following example of existing tests with similar setup.

**Comment by madmir** [comment]
Thanks for your time and giving opportunities for people to get involved.

**Comment by ahsonkhan** [comment]
> Thanks for your time and giving opportunities for people to get involved.

Absolutely. We definitely welcome and encourage community contributions whether in the JSON space or anything in this repo. Feel free to ping me on other PRs of yours and I can help review them or help where possible.

There are quite a few up-for-grabs issues: https://github.com/dotnet/corefx/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs and a few marked as easy as well: https://github.com/dotnet/corefx/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs+label%3Aeasy

Specific to the issue you were working on, we still have some test gaps to fill :)

**Comment by danmosemsft** [comment]
Seconded @madmir, we invite you to be a regular contributor!

**Pull request merged**